### ISSUE TITLE
Rename file: Graphical issue when newline is part of filename

### ISSUE THREAD

**Issue by Piiit** [linuxmint/nemo#1179]
When I rename a file with a copy/paste filename that contains a newline, the list view of nemo creates a big gap between each file, not only the renamed one. In addition, the detail-view does not show the correct size of its icon.

First image: No newline character in filename...
![auswahl_007](https://cloud.githubusercontent.com/assets/6096063/15385156/edf3b334-1d9f-11e6-9e2e-51f7c8035552.jpg)

Second image: detail view with too small icon...
![auswahl_009](https://cloud.githubusercontent.com/assets/6096063/15385150/e370b092-1d9f-11e6-9c42-e7d0b061e4cf.jpg)

Third image: Too large spacing...
![auswahl_008](https://cloud.githubusercontent.com/assets/6096063/15385154/ead0f68a-1d9f-11e6-943a-dacfd3c6a72b.jpg)

**Comment by xenopeek** [comment]
What Linux distribution were you using and what version of Nemo?

Older versions of Nemo (current is 3.0.6) might depend on GTK+ patches for the treeview that might not be installed on your distro. Current version of Nemo no longer needs that so should work fine.

I can't reproduce this issue on Linux Mint 18. Newline is displayed in all views and spacing and icon size are consistent.

**Comment by Piiit** [comment]
@xenopeek I use Nemo 3.0.6, it is still an issue.

Steps to reproduce this issue:
1. Create a file in nemo: `test.txt`
2. Open this test.txt file with xed, create 3 lines separated with newline 
```
test
test
test
```
3. Copy these lines and go back to nemo, rename the file `test.txt` by pasting the previously copied lines into the filename
4. All Icons get huge gaps between them.

**Comment by JosephMcc** [comment]
I'm not familiar with nemos view code but I would imagine the "equal sizing" that is going on here is intentional. I'm not sure how normal it is for people to save filenames with newlines in them.

**Comment by Piiit** [comment]
Ok, then let me rephrase this issue into a feature request:
Remove newline and tab chars from filenames when copy and pasting them from source text with newlines.

**Comment by xenopeek** [comment]
This doesn't happen in icon view nor in list view. It happens only in compact view.

Looking at other file managers:
- Caja works exactly like Nemo.
- Thunar shows the unicode number glyph ([000A]) in place of the life feed. In list view and compact view showing the filename on one line. In icon view still showing the unicode number glyph but also showing the filename on three lines.
- Dolphin just discards control characters and doesn't show anything in their place (e.g., it just shows the filename as "asdfasdfasdfasdfasdfasdf.txt" regardless of the view).
- Nautilus works exactly like Nemo in icon view. In list view it does the same thing as Thunar. It doesn't have compact view.

**Comment by JosephMcc** [comment]
So basically there is no standard behavior :)

**Comment by Piiit** [comment]
@JosephMcc @xenopeek  
I think you forgot to remove the label for "no response" here, and add feature request as requested in https://github.com/linuxmint/nemo/issues/1179#issuecomment-249661418... 

Or is it unclear what behaviour we should implement? Removing newlines and other whitespace characters (except space) seems the best way for me. What do you think?

**Comment by dg1727** [comment]
Being able to put a newline into a filename using Nemo can be useful in quick testing of user apps that access files.  Maybe there can also be a preferences option so that either whitespace is removed on file rename [as Piiit suggests], or (if the user so elects) only a confirmation dialog is shown if the user enters a filename that contains a newline.  

Furthermore, Nemo still has to be able to show file-lists that contain filenames with newlines, since a bug in some user-written app could generate such filenames at any time.  If such a user bug then causes Nemo to do what's shown in Piiit's screenshot with the green folder icons, then that adds insult to insult:  not only did the user's code introduce a newline, but also not as many icons are now visible in the Nemo window, and it can make Nemo look not-well-written if Nemo didn't handle that case well.  

Discarding newlines as Dolphin does (xenopeek's comment) is probably not a good solution, since an end-user who is a software developer might not realize that a newline filename can be created by her/his code in some situation until the file manager (Nemo) shows it to her/him.  This also means that Piiit's option of 
> Just not displaying [newline and tab chars] until a rename procedure is started 

might not be preferable.

**Comment by icarter09** [comment]
Many thanks for submitting your feature request for Nemo. Your feature request has been documented and will be reviewed by the Linux Mint team.

For more information on our workflow and feature requests, read https://linuxmint-troubleshooting-guide.readthedocs.io/en/latest/faq.html.
